src/interfaces.jl

```diff
@@ -110,3 +122,27 @@ function reverse_time(sol::Solver)
     options.stopevent = (t,y)->stopevent(2*t0-t,y)
     return solve(ode_reversed,stepper,options)
 end
+
+
+
+# The the elements of tspan should basically be scalars and support
+# most of the scalar operations.  In particular the element type
+# should be closed under the division.
+function make_consistent{S<:AbstractStepper}(y0, tspan::AbstractVector, stepper::Type{S})
+    t_test = 1/(tspan[end]-tspan[1])
+    Tt = typeof(t_test)
+    t_new = convert(Vector{Tt},tspan)
+
+    y_test = y0./t_test
+    Ty = typeof(y_test)
+    if typejoin(Ty,AbstractArray) == AbstractArray
+        Ey = promote_type(map(typeof,y_test)...)
```

Comment:
I kind of expected bad performance, the gain is that it converts the `Any[1,BigFloat(1)]` to `BigFloat[1,1]`.  With your code it would leave it as `Any`.  I'm good either way, I just wanted to point out that it's a trade off we should be conscious about.  Honestly I believe that people who are using the top level interface won't be putting huge arrays into it.  How about we test if `y_test` is of a concrete type before trying to figure out the type with `promote_type`
```
if Ty <: AbstractArray
        if isleaftype(eltype(Ty)) & (eltype(Ty) <: AbstractFloat)
            y_new = convert(Ty,y0)
        else
            Ey = promote_type(map(typeof,y_test)...)
            y_new = copy!(similar(y0,Ey),y0)
        end
    else
        y_new = convert(Ty,y0)
    end
```